source/docs/software/commandbased/commands-v3/migration-from-v2.rst

```diff
@@ -0,0 +1,578 @@
+# Migrating from Commands v2 to v3
+
+This guide helps teams migrate existing Commands v2 code to Commands v3. It covers the key conceptual changes, provides side-by-side code comparisons, and highlights common pitfalls.
+
+.. warning::
+   Commands v3 is **Java-only**. If you need C++ or Python support, continue using Commands v2.
+
+## Should You Migrate?
+
+**WPILib recommends migrating to Commands v3 for Java teams.** Commands v3 is the future direction of command-based programming with ongoing development and new features including improved telemetry, enhanced triggers, and self-canceling commands.
```

Comment:
Similar to my previous comment, I think this needs to be reworked to not be so final on a framework that doesn't have much runtime